internal/controlplane/handlers_authz.go

```diff
@@ -632,23 +627,134 @@ func (s *Server) UpdateRole(ctx context.Context, req *minder.UpdateRoleRequest)
 		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
 	}
 
-	// We may be given a human-readable identifier which can vary over time. Resolve
-	// it to an IDP-specific stable identifier so that we can support subject renames.
-	identity := &auth.Identity{
-		Provider:  nil,
-		UserID:    sub,
-		HumanName: sub,
+	// Validate the subject and email - decide if it's about updating an invitation or a role assignment
+	if sub == "" && email != "" {
+		if flags.Bool(ctx, s.featureFlags, flags.UserManagement) {
+			return s.updateInvite(ctx, targetProject, authzRole, email)
+		}
+		return nil, util.UserVisibleError(codes.Unimplemented, "user management is not enabled")
+	} else if sub != "" && email == "" {
+		// If there's a subject, we assume it's a role assignment update
+		return s.updateRole(ctx, targetProject, authzRole, sub)
+	}
+	return nil, util.UserVisibleError(codes.InvalidArgument, "one of subject or email must be specified")
+}
+
+func (s *Server) updateInvite(
+	ctx context.Context,
+	targetProject uuid.UUID,
+	authzRole authz.Role,
+	email string,
+) (*minder.UpdateRoleResponse, error) {
```

Comment:
I wish this could be an `upsert`... I'm assuming that doesn't make sense given our RPCs, though, right?